### DISCUSSION TITLE
How to create custom database models

### DISCUSSION THREAD

**Discussion by CarlosLoboxyz** [pocketbase/pocketbase#3655] [Q&A]
I came from an python / flask / sqlalchemy background, back there i remember using classes to create custom models and relationships, thus making migrations a lot easier, i was wondering that if i make a local app and then want to recreate the models i have in local in some server then i would have to manually go into the pocketbase dashboard and create this things again.

How can i create models and relationships (using code) that i would then migrate to a server if i'm using the javascript sdk?

I've read [this discussion](https://github.com/pocketbase/pocketbase/discussions/2697#discussioncomment-6154067) where they recommend [using go to extend your models](https://pocketbase.io/docs/go-custom-models/) but wouldn't that mean that i have to recompile the entire pocketbase binary in order to have custom models?

**Comment by ganigeorgiev** [comment]
> if i make a local app and then want to recreate the models i have in local in some server then i would have to manually go into the pocketbase dashboard and create this things again.

No. For that purposes you can use the builtin migrations ([Go](https://pocketbase.io/docs/go-migrations/) or [JS](https://pocketbase.io/docs/js-migrations/)).

If you use the prebuilt executable, automigrate is enabled by default so to "transfer" the local collections to the remote is enough to copy the `pb_migrations` directory to your server and restart the application. The unapplied migrations will be automatically executed on `serve`.

**Reply by CarlosLoboxyz** [reply]
well, that's really nice for the example above thanks.
